gh-71566: Fix SocketType introduction in the socket documentation. (#93288)
python/cpython: Doc/library/socket.rst

.. data:: SocketType

>    This is the base class for a type object of socket objects, and
>    ``isinstance(socket(...), SocketType)`` returns ``True``.

Reviewer: ```suggestion
   This is the base class for socket objects, such that
   ``isinstance(socket(...), SocketType)`` is ``True``.
```

Author: Thank you for your professional advice. As we can see, `socket.SocketType` is the base class for the type object `socket.socket`. Consider that elsewhere in this document `socket object` refers to an instance of the type object `socket.socket`. Wouldn't it be less precise if we just said that `SocketType` is the base class of `socket objects`? Looking forward to your more suggestions.

   .. versionchanged:: 3.0

>       SocketType is now the base class for the socket object type.

Reviewer: ```suggestion
      SocketType is the base class for socket objects 
      rather than being ``type(socket(...))``` directly.
```

.. data:: SocketType

   This is the base class for a type object of socket objects, such that
   ``isinstance(socket(...), SocketType)`` is ``True``.

   .. versionchanged:: 3.0
      SocketType is the base class for the socket object type rather than

>       being ``type(socket(...))``` directly.

Reviewer 2: There is a RST marker error:

```suggestion
      being ``type(socket(...))`` directly.
```
